components/docusign/sources/envelope-sent-or-complete/envelope-sent-or-complete.js

```diff
@@ -0,0 +1,73 @@
+const docusign = require("../../docusign.app.js");
+
+module.exports = {
+  key: "docusign-envelope-sent-or-complete",
+  name: "Envelope Sent or Complete",
+  description:
+    "Emits an event when an envelope status is set to sent or complete",
+  version: "0.0.2",
+  dedupe: "unique",
+  props: {
+    docusign,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+    account: {
+      propDefinition: [
+        docusign,
+        "account",
+      ],
+    },
+  },
+  methods: {
+    _getLastEvent() {
+      return this.db.get("lastEvent");
+    },
+    _setLastEvent(lastEvent) {
+      this.db.set("lastEvent", lastEvent);
+    },
+    monthAgo() {
+      const monthAgo = new Date();
+      monthAgo.setMonth(monthAgo.getMonth() - 1);
+      return monthAgo;
+    },
+    generateMeta({
+      envelopeId: id, emailSubject: summary,
+    }, ts) {
+      return {
+        id,
+        summary,
+        ts,
+      };
+    },
+  },
+  async run(event) {
+    const { timestamp: ts } = event;
+    const lastEvent = this._getLastEvent() || this.monthAgo().toISOString();
+    const baseUri = await this.docusign.getBaseUri(this.account);
+    let done = false;
+    const params = {
+      from_date: lastEvent,
+      status: "sent,completed",
```

Comment:
Missed this the first time through, but this needs to be a prop. Allow selecting either `sent` (default) or `completed`.

Reply:
Updated